Travel Scheduler/View Controllers/Place.m

```diff
@@ -48,37 +51,42 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary {
             self.isSelected = NO;
             [self makeScheduleDictionaries];
         }
-    });
     return self;
 }
 
-- (instancetype)initWithName:(NSString *)name withCompletion:(void (^)(bool sucess, NSError *error))completion {
-    __block Place *place;
+- (instancetype)initWithName:(NSString *)name beginHub:(bool)isHub{
+    self = [super init];
+    dispatch_semaphore_t didCreatePlace = dispatch_semaphore_create(0);
     [[APIManager shared]getCompleteInfoOfLocationWithName:name withCompletion:^(NSDictionary *placeInfoDictionary, NSError *error) {
         if(placeInfoDictionary) {
-            place = [place initWithDictionary:placeInfoDictionary];
-            completion(YES, nil);
+            [self initWithDictionary:placeInfoDictionary];
+            if(isHub) {
+                self.isHub = YES;
+                [self createDictionaryOfArrays];
+            }
+            dispatch_semaphore_signal(didCreatePlace);
         }
         else {
             NSLog(@"could not get dictionary");
-            completion(NO, error);
+            dispatch_semaphore_signal(didCreatePlace);
         }
```

Comment:
Instead of repeating dispatch_semaphore_signal(didCreatePlace); inside both if and else, you could just put it after the bracket of this line. Idk, less repeated code?

Reply:
Done!